cli/src/utils/xlr/visitors/plugin.ts

```diff
@@ -0,0 +1,226 @@
+import type { TopLevelNode } from '@player-tools/xlr-utils';
+import { isNodeExported, isTopLevelNode } from '@player-tools/xlr-utils';
+import path from 'path';
+import ts from 'typescript';
+import fs from 'fs';
+import type { Manifest, NamedType, RefNode } from '@player-tools/xlr';
+import type { TsConverter } from '@player-tools/xlr-converters';
+import type { VisitorProps } from './types';
+import { PLAYER_PLUGIN_INTERFACE_NAME } from '../consts';
+
+/**
+ * Follows references to get the actual underlying declaration
+ */
+function getUnderlyingNode(
+  element: ts.TypeQueryNode | ts.TypeReferenceNode,
+  checker: ts.TypeChecker
+): TopLevelNode | undefined {
+  let referencedSymbol;
+  if (ts.isTypeReferenceNode(element)) {
+    referencedSymbol = checker.getSymbolAtLocation(element.typeName);
+  } else {
+    referencedSymbol = checker.getSymbolAtLocation(element.exprName);
+  }
+
+  const alias = referencedSymbol
+    ? checker.getAliasedSymbol(referencedSymbol)
+    : undefined;
+  let varDecl;
+  /**
+   * The TypeChecker will return the interface/type declaration or the variable statement
+   * so if we are getting a variable, we need to grab the variable declaration
+   */
+  if (ts.isTypeReferenceNode(element)) {
+    varDecl = alias?.declarations?.[0];
+  } else {
+    varDecl = alias?.declarations?.[0].parent.parent;
+  }
+
+  if (varDecl && isTopLevelNode(varDecl)) {
+    return varDecl;
+  }
+}
+
+/**
+ * Fixes ExpressionHandler ref nodes that don't have argument names because some way of writing ExpressionHandler prevent the ts compiler from inferring them
+ */
+function fixExpressionArgNames(
```

Comment:
That's something I was considering. I would still say we should keep them as references to the ExpressionHandler type since that is the most accurate way of describing it. Putting it into a more uniform type might be a good idea though to make parsing easier.

Reply:
Maybe we just do that as part of the sdk?

Reply:
Yeah, it would work well as a Transform we provide in the Language Service for the SDK to run on load.